presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Memo.java

```diff
@@ -125,10 +129,31 @@ public PlanNode replace(int group, PlanNode node, String reason)
         incrementReferenceCounts(node, group);
         getGroup(group).membership = node;
         decrementReferenceCounts(old, group);
+        evictStatistics(group);
 
         return node;
     }
 
+    private void evictStatistics(int group)
+    {
+        getGroup(group).stats = null;
+        for (int parentGroup : getGroup(group).references.elementSet()) {
+            if (parentGroup != ROOT_GROUP_REF) {
+                evictStatistics(parentGroup);
+            }
+        }
+    }
+
+    public Optional<PlanNodeStatsEstimate> getStats(int group)
+    {
+        return Optional.ofNullable(getGroup(group).stats);
+    }
+
+    public void storeStats(int group, PlanNodeStatsEstimate stats)
+    {
+        getGroup(group).stats = requireNonNull(stats, "stats is null");
```

Comment:
in case of `stats` was not null before calling `storeStats`, should not we `evictStatistics`?